lib/publiccloud/provider.pm

```diff
@@ -0,0 +1,64 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Base helper class for public cloud
+#
+# Maintainer: Clemens Famulla-Conrad <cfamullaconrad@suse.de>
+
+package publiccloud::provider;
+use Mojo::Base -base;
+
+has key_id     => undef;
+has key_secret => undef;
+has region     => undef;
+has prefix     => 'openqa';
+
+=head1 METHODS
+ 
+=head2 init
+ 
+Needs provider specific credentials, e.g. key_id, key_secret, region.
+
+=cut
+sub init {
+    die('init() isn\'t implemented');
+}
+
+
+=head2 find_img
+
+Retrieves the image-id by given image C<name>.
+
+=cut
+sub find_img {
+    die('find_image() isn\'t implemented');
```

Comment:
I am not sure here , but isn't this code will be never ever executed ? I mean you already dying in int or as usual there always a way to do it if you really want ?

Reply:
The intention was, if someone use a implementation (e.g. google cloud), which doesn't provide this method, will be informed. It's like a abstract method.
I could just return undef, so a caller would never get a image. But then I do not have a difference between not implemented and image not found.
I would vote for that not implemented approach.